first_set_of_refactorings/extract_variable/example.rb

```diff
@@ -0,0 +1,52 @@
+# frozen_string_literal: true
+
+# formerly: Introduce Explaining Variables
+# inverse of: Inline Variable
+# @see https://memberservices.informit.com/my_account/webedition/9780135425664/html/extractfunction.html
+
+# Goal: Add a name to an expression
+
+module FirstSetOfRefactorings
+  module ExtractVaiable
+    module Example
+      class BeforeRefactor < RefactorBase
+        def price(order)
+          # price is base price - quantity discount + shipping
+          return order.quantity * order.item_price -
+              [0, order.quantity - 500].max * order.item_price * 0.05 +
+              [order.quantity * order.item_price * 0.1, 100.0].min
+        end
+      end
+
+      class Refactor1 < RefactorBase
+        def price(order)
+          # price is base price - quantity discount + shipping
+          base_price = order.quantity * order.item_price
+          return base_price -
+              [0, order.quantity - 500].max * order.item_price * 0.05 +
+              [base_price * 0.1, 100.0].min
+        end
+      end
+
+      class Refactor2 < RefactorBase
+        def price(order)
+          # price is base price - quantity discount + shipping
+          base_price = order.quantity * order.item_price
+          quantity_discount = [0, order.quantity - 500].max *
+            order.item_price * 0.05
+          return base_price - quantity_discount + [base_price * 0.1, 100.0].min
+        end
+      end
+
+      class Refactor3 < RefactorBase
+        def price(order)
+          # price is base price - quantity discount + shipping
```

Comment:
Comment not required in the last example because now your final statement says this